src/curated-corpus/components/actions/DismissProspectAction/DismissProspectAction.test.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { MockedProvider, MockedResponse } from '@apollo/client/testing';
+import { render, screen, waitFor } from '@testing-library/react';
+import { SnackbarProvider } from 'notistack';
+import { DismissProspectAction } from './DismissProspectAction';
+import {
+  successMock,
+  errorMock,
+  errorMessage,
+} from '../../../integration-test-mocks/dismissProspect';
+import userEvent from '@testing-library/user-event';
+import { apolloCache } from '../../../../api/client';
+import { getTestProspect } from '../../../helpers/prospects';
+
+describe('The DismissProspectAction', () => {
+  const prospectId = getTestProspect().id;
+  const onDismissProspect = jest.fn();
+
+  const renderComponent = (mocks?: MockedResponse[]) => {
+    render(
+      <MockedProvider mocks={mocks} cache={apolloCache}>
+        <SnackbarProvider maxSnack={1}>
+          <DismissProspectAction
+            prospectId={prospectId}
+            onDismissProspect={onDismissProspect}
+          />
+        </SnackbarProvider>
+      </MockedProvider>
+    );
+  };
+
+  it('should render the dismiss button and call the onDismissProspect function without an error message', async () => {
+    renderComponent([successMock]);
+
+    const dismissBtn = screen.getByTestId('dismissButton');
+
+    expect(dismissBtn).toBeInTheDocument();
+
+    userEvent.click(dismissBtn);
+
+    await waitFor(() => expect(onDismissProspect).toHaveBeenCalledTimes(1));
+
+    // asserting that onDismissProspect is called with the correct arguments - error message should be undefined for a successful mutation call
+    await waitFor(() =>
+      expect(onDismissProspect).toHaveBeenCalledWith(prospectId, undefined)
+    );
+  });
+
+  /**
+   * Note: this test *should* work but doesn't. There are past and open GitHub issues
+   * about it on the Apollo Client project. Unskip this test when the promised overhaul
+   * of the Mocked Provider materialises.
+   *
+   * https://github.com/apollographql/apollo-client/issues/4283 (closed)
+   * https://github.com/apollographql/apollo-client/issues/7167 (open since 2020).
+   *
+   */
+  xit('should render the dismiss button and call the onDismissProspect function with an error message', async () => {
```

Comment:
This looks familiar - does this issue here, too? Should this comment block simply reference the explanation already elsewhere in the database instead of copying it? 

Also, I'm not sure what the skipped test is meant to assert - could the description be updated so that it's clearer?

Reply:
Adding a comment inside the test!